src/charm.py

```diff
@@ -160,6 +161,26 @@ def _configure(self, _):
 
         self.unit.status = ActiveStatus()
 
+    def _update_status(self, _):
+        """Update the status of the charm."""
+        # Run configure to reset the status
+        self._configure(_)
+
+        if not isinstance(self.unit.status, ActiveStatus):
+            # Status is already not active so do nothing.
+            return
+
+        external_url = urlparse(self._external_url)
+        prometheus_server = Prometheus(web_route_prefix=external_url.path)
+
+        if not prometheus_server.is_healthy():
+            self.unit.status = BlockedStatus("Prometheus failed health check")
+            return
```

Comment:
Should we have some more serious side effects here?
Unless we rely on self-monitoring?